packages/roosterjs-content-model/lib/publicTypes/format/ContentModelWithDataset.ts

```diff
@@ -0,0 +1,11 @@
+import { DatasetFormat } from './formatParts/DatasetFormat';
+
+/**
+ * Represents base format of an element that supports dataset and/or metadata
+ */
+export interface ContentModelWithDataset<T> {
```

Comment:
Question, is the generic type going to be used later?

Reply:
Yes, when parse metadata, we want this type to help us prevent unwanted type mismatch.